### ISSUE TITLE
(Zed extension) slint-lsp not recognized when using nix flake and direnv

### ISSUE THREAD

**Issue by manuel-plavsic** [slint-ui/slint#9826]
### Bug Description

When I use direnv (default settings OOTB, i.e. `"load_direnv": "direct"`), the Zed slint extension cannot find the right binary. The right binaries are detected for all other extensions (e.g. `rust-analyzer`, `json-language-server`, ...).

I don't have a global version of `rust-analyzer` or `json-language-server`. They are only defined in the project's flake.

The exact error is:

```
Language server slint:

initializing server slint, id 4: server shut down
-- stderr --
Could not start dynamically linked executable: /home/manuel/.local/share/zed/extensions/work/slint/slint-lsp-v1.13.1/slint-lsp/slint-lsp
NixOS cannot run dynamically linked executables intended for generic
linux environments out of the box. For more information, see:
https://nix.dev/permalink/stub-ld
```

My current workaround is:

```
  "lsp": {
    "slint": {
      "binary": {
        "path": "/nix/store/2767m8rrbjs2fww2a6vrdvd2dj8inym3-slint-lsp-1.13.1/bin/slint-lsp", // update whenever the flake gets updated
        "arguments": [],
        "env": {}
      }
    }
  }
```

However, I cannot really add this to my local `.zed/settings.json`, as the path on my macOS/darwin system is different: it is `/nix/store/lf74g0wjdbk4v26d8k0k75f7vyk1x8fx-slint-lsp-1.13.1/bin/slint-lsp`. 

Moving the config above to my global Zed settings would not make much sense, because then I would have to constantly change it when working on different projects.

In my opinion, the correct solution would be to evaluate the right binary in this order:
- use the slint-lsp binary defined in the Zed settings
- use the first found slint-lsp binary in the `$PATH` (the new functionality)
- fallback to `~/.local/share/zed/extensions/work/slint/slint-lsp-v1.13.1/slint-lsp/slint-lsp`

### Reproducible Code (if applicable)

A nix flake with `pkgs.slint-lsp` in `buildInputs`.

### Environment Details

- Slint Version: 1.13.1
- Platform/OS: Nix Flake & NixOS.
- Programming Language: -
- Backend/Renderer: -

**Comment by ogoffart** [comment]
We remove the check for slint-lsp in path in https://github.com/slint-ui/slint/pull/9271/files#diff-903c5c4a385f4aa74b5c00528ce7c79bd636717be31c600dfb99eb4a94c644edL33

IIRC the reason is that we wanted the extension to always use the right version of slint-lsp if not explicitly configured.

But I guess you're arguing that checking in the PATH before using the downloaded version is always the right thing to do?

**Comment by manuel-plavsic** [comment]
I see, your code prevents using the wrong version of `slint-lsp`.

The problem is that NixOS cannot run dynamically linked executables intended for generic Linux environments. However, I also understand why you would want to solely rely on the binary provided by the extension: it is just gonna work on all platform and most Linux distros.

I think that other Zed extensions also provide a binary, but it is used only as a fallback. They check for what is in the `PATH` first. For this reason, I would argue that checking there first is indeed always the right thing to do.

I would consider adding the following flag (maybe you can think of a better name):

```
 "lsp": {
    "slint": {
      "binary": {
        "path_allowed": true|false
      }
    }
  }
```

If it is `true`, then the path is checked and has precedence. If the value is `false`, then your intended behavior is kept. I would argue that the default value should be `true` here (this way the extension behaves similarly to the other extensions), but it is possible to ignore the path by setting it to `false` (to solve or prevent all sorts of version compatibility issues).

**Comment by tronical** [comment]
> The right binaries are detected for all other extensions (e.g. rust-analyzer, json-language-server, ...).

@manuel-plavsic I'm curious, how does zed or the corresponding extension pick the rust-analyzer binary?

(with vs code it defaults to the bundled binary, similar to what we do right now now)

**Comment by manuel-plavsic** [comment]
> > The right binaries are detected for all other extensions (e.g. rust-analyzer, json-language-server, ...).
> 
> [@manuel-plavsic](https://github.com/manuel-plavsic) I'm curious, how does zed or the corresponding extension pick the rust-analyzer binary?
> 
> (with vs code it defaults to the bundled binary, similar to what we do right now now)

The Zed extensions do not work with the bundled binaries. I explicitly included the `rust-analyzer` binary in my PATH (the nix package `rust-analyzer` is defined in my nix flake). The same is true for other extensions and their nixpkgs counterpart, e.g., `package-version-server`, `nil`, `nixd`, `texlab`, etc.

**Comment by tronical** [comment]
Ok, let me see if I understand correctly: You're using Zed's builtin-in rust-analyser support, which by default picks up the rust-analyzer binary from your `PATH` (which you have set up accordingly). Is that correct?

**Comment by manuel-plavsic** [comment]
@tronical that is correct: the binary in the PATH has priority over the one provided by the built-in `rust-analyzer` support. See my comment in the related PR.
